src/common/yaml_config_info.h

```diff
@@ -0,0 +1,50 @@
+#include "fam_config_info.h"
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+#include <yaml-cpp/yaml.h>
+
+using namespace std;
+class yaml_config_info : public config_info {
+  public:
+    void config_init(std::string file_name);
+    std::string get_key_value(std::string key);
+    std::vector<std::string> get_value_list(std::string key);
+    int get_value_type(std::string key);
+
+  private:
+    YAML::Node config;
+};
+
+void yaml_config_info::config_init(std::string file_name) {
+    config = YAML::LoadFile(file_name);
+}
+std::string yaml_config_info::get_key_value(std::string key) {
+    std::string value;
+    if (config[key]) {
+        value = (config[key].as<std::string>());
+        return value;
+    } else {
+        cout << key << " key not found " << endl;
+    }
+}
+std::vector<std::string> yaml_config_info::get_value_list(std::string key) {
+    std::vector<std::string> valueList;
+    if (config[key].IsSequence()) {
+        for (auto element : config[key]) {
+            valueList.push_back(element.as<std::string>());
+        }
+        return valueList;
+    } else {
+        if (!config[key]) {
+            cout << key << " key not found." << endl;
```

Comment:
return appropriate error values. Key not found can be a common use case. so we don't need to print message